File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/semesterService.js
import api from "./api";

const BASE_URL = "/semesters";

const semesterService = {

    /*
     * Get All Semesters
     */
    getAllSemesters: async () => {

        const response = await api.get(BASE_URL);

        return response.data;

    },

    /*
     * Get Semester By Id
     */
    getSemesterById: async (id) => {

        const response = await api.get(`${BASE_URL}/${id}`);

        return response.data;

    },

    /*
     * NEW
     * Get Semesters By Course
     */
    getSemestersByCourse: async (courseId) => {

        const response = await api.get(
            `${BASE_URL}/course/${courseId}`
        );

        return response.data;

    },

    /*
     * Create Semester
     */
    createSemester: async (semester) => {

        const response = await api.post(
            BASE_URL,
            semester
        );

        return response.data;

    },

    /*
     * Update Semester
     */
    updateSemester: async (id, semester) => {

        const response = await api.put(
            `${BASE_URL}/${id}`,
            semester
        );

        return response.data;

    },

    /*
     * Delete Semester
     */
    deleteSemester: async (id) => {

        const response = await api.delete(
            `${BASE_URL}/${id}`
        );

        return response.data;

    }

};

export default semesterService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/SemesterService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.semester.SemesterRequest;
import com.timetable.dto.semester.SemesterResponse;

public interface SemesterService {

    /**
     * Save Semester
     */
    SemesterResponse saveSemester(SemesterRequest request);

    /**
     * Update Semester
     */
    SemesterResponse updateSemester(Long semesterId,
                                    SemesterRequest request);

    /**
     * Delete Semester
     */
    void deleteSemester(Long semesterId);

    /**
     * Get Semester By ID
     */
    SemesterResponse getSemesterById(Long semesterId);

    /**
     * Get All Semesters
     */
    List<SemesterResponse> getAllSemesters();

}